src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -513,6 +513,18 @@ private static bool NeedsSeparator(SyntaxToken token, SyntaxToken next)
                 }
             }
 
+            // No space before an asterisk that's part of a PointerTypeSyntax.
+            if (next.IsKind(SyntaxKind.AsteriskToken) && next.Parent is PointerTypeSyntax)
+            {
+                return false;
+            }
+
+            // The last asterisk of a pointer declaration should be followed by a space.
+            if (token.IsKind(SyntaxKind.AsteriskToken) && next.IsKind(SyntaxKind.IdentifierToken))
```

Comment:
this would fire for other scenarios.  like `x * y` multiplication.  If this is for pointers, then we need to check that that's what we have.